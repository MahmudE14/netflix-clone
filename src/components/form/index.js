import { Container, Base, Error, Title, Text, TextSmall, Link, Input, Submit } from './styles/form'

export default function Form({ children, ...restProps }) {
    return (
        <Container {...restProps}>{children}</Container>
    )
};

Form.Base = function FormContainer({ children, ...restProps }) {
    return <Base { ...restProps }>{children}</Base>
}

Form.Error = function FormContainer({ children, ...restProps }) {
    return <Error { ...restProps }>{children}</Error>
}

Form.Title = function FormContainer({ children, ...restProps }) {
    return <Title { ...restProps }>{children}</Title>
}

Form.Text = function FormContainer({ children, ...restProps }) {
    return <Text { ...restProps }>{children}</Text>
}

Form.TextSmall = function FormContainer({ children, ...restProps }) {
    return <TextSmall { ...restProps }>{children}</TextSmall>
}

Form.Link = function FormContainer({ children, ...restProps }) {
    return <Link { ...restProps }>{children}</Link>
}

Form.Input = function FormContainer({ children, ...restProps }) {
    return <Input { ...restProps }>{children}</Input>
}

Form.Submit = function FormContainer({ children, ...restProps }) {
    return <Submit { ...restProps }>{children}</Submit>
}
