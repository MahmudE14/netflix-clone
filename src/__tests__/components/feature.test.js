import { render, screen } from "@testing-library/react";
import { Feature } from "../../components";

describe("<Feature />", () => {
  it("renders the <Footer /> with populated data", () => {
    const { container } = render(
      <Feature>
        <Feature.Title>Unlimited films, TV programmes and more.</Feature.Title>
        <Feature.SubTitle>Watch anywhere. cancel at any time.</Feature.SubTitle>
      </Feature>
    );

    expect(screen.getByText('Unlimited films, TV programmes and more.')).toBeTruthy()
    expect(screen.getByText('Watch anywhere. cancel at any time.')).toBeTruthy()
    expect(container.firstChild).toMatchSnapshot()
  });

  it("renders the <Footer /> with just a title", () => {
    const { container } = render(
      <Feature>
        <Feature.Title>Unlimited films, TV programmes and more.</Feature.Title>
      </Feature>
    );

    expect(screen.getByText('Unlimited films, TV programmes and more.')).toBeTruthy()
    expect(screen.queryByText('Watch anywhere. cancel at any time.')).toBeFalsy()
    expect(container.firstChild).toMatchSnapshot()
  });

  it("renders the <Footer /> with just a subtitle", () => {
    const { container } = render(
      <Feature>
        <Feature.SubTitle>Watch anywhere. cancel at any time.</Feature.SubTitle>
      </Feature>
    );

    expect(screen.queryByText('Unlimited films, TV programmes and more.')).toBeFalsy()
    expect(screen.getByText('Watch anywhere. cancel at any time.')).toBeTruthy()
    expect(container.firstChild).toMatchSnapshot()
  });
});
