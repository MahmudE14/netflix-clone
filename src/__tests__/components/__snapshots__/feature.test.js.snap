// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Feature /> renders the <Footer /> with just a subtitle 1`] = `
<div
  class="feature__Container-sc-19zs3jp-0 fJSYSJ"
>
  <h2
    class="feature__SubTitle-sc-19zs3jp-2 dMGKTO"
  >
    Watch anywhere. cancel at any time.
  </h2>
</div>
`;

exports[`<Feature /> renders the <Footer /> with just a title 1`] = `
<div
  class="feature__Container-sc-19zs3jp-0 fJSYSJ"
>
  <h1
    class="feature__Title-sc-19zs3jp-1 frMQVC"
  >
    Unlimited films, TV programmes and more.
  </h1>
</div>
`;

exports[`<Feature /> renders the <Footer /> with populated data 1`] = `
<div
  class="feature__Container-sc-19zs3jp-0 fJSYSJ"
>
  <h1
    class="feature__Title-sc-19zs3jp-1 frMQVC"
  >
    Unlimited films, TV programmes and more.
  </h1>
  <h2
    class="feature__SubTitle-sc-19zs3jp-2 dMGKTO"
  >
    Watch anywhere. cancel at any time.
  </h2>
</div>
`;
